src/d3a/d3a_core/export.py

```diff
@@ -169,48 +174,106 @@ def move_root_plot_folder(self):
             shutil.move(os.path.join(old_dir, si), self.plot_dir)
         shutil.rmtree(old_dir)
 
-    def _export_area_with_children(self, area: Area, directory: dir, is_first: bool = False):
+    def _export_spot_markets_stats(self, area: Area, directory: dir, is_first: bool) -> None:
+        """Export bids, offers, trades, statistics csv-files for all spot markets."""
+        self._export_area_stats_csv_file(area, directory, PastDataEnum.SPOT_MARKET, is_first)
+        if not area.children:
+            return
+        self._export_trade_csv_files(past_markets=area.past_markets,
+                                     directory=directory,
+                                     filename=f"{area.slug}-trades",
+                                     is_first=is_first)
+
+        self._export_area_offers_bids_csv_files(past_markets=area.past_markets,
+                                                market_member="offer_history",
+                                                directory=directory,
+                                                filename=f"{area.slug}-offers",
+                                                labels=("slot",) + Offer.csv_fields(),
+                                                is_first=is_first)
+
+        self._export_area_offers_bids_csv_files(past_markets=area.past_markets,
+                                                market_member="bid_history",
+                                                directory=directory,
+                                                filename=f"{area.slug}-bids",
+                                                labels=("slot",) + Bid.csv_fields(),
+                                                is_first=is_first)
+
+    def _export_settlement_markets_stats(self, area: Area, directory: dir, is_first: bool) -> None:
+        """Export bids, offers, trades, statistics csv-files for all settlement markets."""
+        if not ConstSettings.SettlementMarketSettings.ENABLE_SETTLEMENT_MARKETS:
+            return
+        self._export_area_stats_csv_file(area, directory, PastDataEnum.SETTLEMENT_MARKET, is_first)
+        if not area.children:
+            return
+        self._export_trade_csv_files(
+            past_markets=list(area.past_settlement_markets.values()),
```

Comment:
The code is basically the same for all `_export_*_markets_stats()` methods. However, you pass the markets in different ways to the export CSV methods:
```python
# spot markets
past_markets=area.past_markets
# balancing markets
past_markets=area.past_balancing_markets
# settlement markets
past_markets=list(area.past_settlement_markets.values())  # <-- different!
```
As far as I see in `AreaMarkets`, there is no difference in the type of structure of these variables: https://github.com/gridsingularity/d3a/blob/0368635e2bb26db3d5fadb64efff0d6e20ea190d/src/d3a/models/area/markets.py#L45-L47
So why can't we just use the following?
```python
past_markets=area.past_settlement_markets
```

Reply:
This is because `area.past_settlement_markets -> Dict` and ´area.past_markets -> List´ . Have a look into the are class. I discussed this briefly with Spyros and we decided to leave it as this as we use he dictionary behaviour (time_slots are available) in other parts of the code for the settlement markets.

Reply:
Thanks. I think it would be good to refactor either one or the other (in order to always have the same return type). Maybe another time.